src/app/market/buy/checkout-process/checkout-process.component.ts

```diff
@@ -79,7 +80,17 @@ export class CheckoutProcessComponent implements OnInit, OnDestroy {
 
     this.cartService.list()
       .takeWhile(() => !this.destroyed)
-      .subscribe(cart => this.cart = cart);
+      .subscribe((cart: Cart) => {
+        this.cart = cart
+
+        // no need to reset steper if cart is empty initially.
+        if (this.cart.countOfItems === 0 && !this.isInitialCartItemFetching) {
```

Comment:
We can use `this.cart` to check if it's the initial fetching.

``` 
.subscribe((cart: Cart) => {
        // note: this.cart & cart, two different ones!
        if (this.cart && cart.countOfItems === 0) {
                 this.resetStepper();
        }
        this.cart = cart
        this.cartFormGroup.patchValue({ itemsInCart: this.cart.countOfItems });
} 
```

Still this seems a bit overkil and I don't trust the stepper.reset() I think we can probably simplify this.
Could you write out the issue again? What's exactly

Reply:
Also this.stepper.reset () should set linear to true no?
Why is this.blockGoingBack needed

Reply:
see #1020

Reply:
and `this.stepper.reset ()` set linear false and default.